Source/santad/Logs/EndpointSecurity/Serializers/BasicStringTest.mm

```diff
@@ -1041,6 +1047,132 @@ - (void)testtestSerializeMessageGatekeeperOverride {
+      {ES_TCC_AUTHORIZATION_REASON_MDM_POLICY, "MDM_POLICY"},
+      {ES_TCC_AUTHORIZATION_REASON_SERVICE_OVERRIDE_POLICY, "SERVICE_OVERRIDE_POLICY"},
+      {ES_TCC_AUTHORIZATION_REASON_MISSING_USAGE_STRING, "MISSING_USAGE_STRING"},
+      {ES_TCC_AUTHORIZATION_REASON_PROMPT_TIMEOUT, "PROMPT_TIMEOUT"},
+      {ES_TCC_AUTHORIZATION_REASON_PREFLIGHT_UNKNOWN, "PREFLIGHT_UNKNOWN"},
+      {ES_TCC_AUTHORIZATION_REASON_ENTITLED, "ENTITLED"},
+      {ES_TCC_AUTHORIZATION_REASON_APP_TYPE_POLICY, "APP_TYPE_POLICY"},
+      {ES_TCC_AUTHORIZATION_REASON_PROMPT_CANCEL, "PROMPT_CANCEL"},
+      {(es_tcc_authorization_reason_t)1234, "UNKNOWN"},
+  };
+
+  for (const auto &kv : authReasonToString) {
+    XCTAssertCppStringEqual(santa::GetTCCAuthorizationReasonString(kv.first), kv.second);
+  }
+}
+
+- (void)testSerializeMessageTCCModification {
+  es_file_t procFile = MakeESFile("foo");
+  es_process_t proc = MakeESProcess(&procFile, MakeAuditToken(12, 34), MakeAuditToken(56, 78));
+  es_message_t esMsg = MakeESMessage(ES_EVENT_TYPE_NOTIFY_TCC_MODIFY, &proc);
+
+  es_file_t instigatorProcFile = MakeESFile("fooInst");
+  es_process_t instigatorProc =
+      MakeESProcess(&instigatorProcFile, MakeAuditToken(21, 43), MakeAuditToken(65, 87));
+
+  es_file_t responsibleFile = MakeESFile("fooApp");
+  es_process_t responsibleProc =
+      MakeESProcess(&responsibleFile, MakeAuditToken(21, 43), MakeAuditToken(65, 87));
+
+  audit_token_t tokInstigator = MakeAuditToken(654, 321);
+  audit_token_t tokResponsible = MakeAuditToken(111, 222);
+
+  es_event_tcc_modify_t tcc = {
+      .service = MakeESStringToken("SystemPolicyDocumentsFolder"),
+      .identity = MakeESStringToken("security.northpole.santa"),
+      .identity_type = ES_TCC_IDENTITY_TYPE_BUNDLE_ID,
+      .update_type = ES_TCC_EVENT_TYPE_MODIFY,
+      .instigator_token = tokInstigator,
+      .instigator = &instigatorProc,
+      .responsible_token = &tokResponsible,
+      .responsible = &responsibleProc,
+      .right = ES_TCC_AUTHORIZATION_RIGHT_SESSION_PID,
+      .reason = ES_TCC_AUTHORIZATION_REASON_SERVICE_POLICY,
+  };
+
+  esMsg.event.tcc_modify = &tcc;
+
+  std::string got = BasicStringSerializeMessage(&esMsg);
+  std::string want =
+      "action=TCC_MODIFICATION|event_type=MODIFY|service=SystemPolicyDocumentsFolder"
+      "|identity=security.northpole.santa|identity_type=BUNDLE_ID|auth_right=SESSION_PID"
+      "|auth_reason=SERVICE_POLICY|event_pid=21|event_ppid=65|event_process=fooInst"
+      "|event_processpath=fooInst|event_uid=-2|event_user=nobody"
+      "|event_gid=-1|event_group=nogroup|pid=12|ppid=56|process=foo|processpath=foo"
+      "|uid=-2|user=nobody|gid=-1|group=nogroup|machineid=my_id\n";
+
+  XCTAssertCppStringEqual(got, want);
+
+  tcc.instigator = NULL;
+  tcc.update_type = ES_TCC_EVENT_TYPE_CREATE;
+  tcc.identity_type = ES_TCC_IDENTITY_TYPE_POLICY_ID;
+  tcc.right = ES_TCC_AUTHORIZATION_RIGHT_ALLOWED;
+  tcc.reason = ES_TCC_AUTHORIZATION_REASON_PROMPT_TIMEOUT;
```

Comment:
I'm sure it doesn't matter as this is just a test, but this combination of right & reason seems unlikely at best.

Reply:
Yeah, definitely not trying to create real data. I already check all the enums, separately, here i'm just randomly picking a smattering of values.